discopop_explorer/pattern_detectors/task_parallelism/preprocessor.py

```diff
@@ -383,6 +179,255 @@ def cu_xml_preprocessing(cu_xml: str) -> str:
+            parent.readPhaseLines.set("count", str(int(parent.readPhaseLines.get("count")) - 1))
+    except TypeError:
+        parent.readPhaseLines._setText(parent_copy.callsNode.nodeCalled.get("atLine"))
+        parent.readPhaseLines.set("count", "1")
+
+    try:
+        if parent_copy.callsNode.nodeCalled.get("atLine") in \
+                parent.writePhaseLines.text:
+            parent.writePhaseLines._setText(parent.writePhaseLines.text.replace(
+                parent_copy.callsNode.nodeCalled.get("atLine") + ",", ""))
+            parent.writePhaseLines._setText(
+                parent.writePhaseLines.text.replace(parent_copy.callsNode.nodeCalled.get("atLine"),
+                                                    ""))
+            parent.writePhaseLines.set("count", str(int(parent.writePhaseLines.get("count")) - 1))
+    except TypeError:
+        parent.writePhaseLines._setText(parent_copy.callsNode.nodeCalled.get("atLine"))
+        parent.writePhaseLines.set("count", "1")
+
+
+def __update_rwi_lines(parent_copy):
+    """Updates instruction/readPhase/writePhase lines of parent_copy
+    :param parent_copy: cu node to be updated"""
+    try:
+        for tmp_line in parent_copy.instructionLines.text.split(","):
+            if not line_contained_in_region(
+                    tmp_line,
+                    parent_copy.get("startsAtLine"),
+                    parent_copy.get("endsAtLine")):
+                parent_copy.instructionLines._setText(
+                    parent_copy.instructionLines.text.replace(tmp_line + ",", ""))
+                parent_copy.instructionLines._setText(
+                    parent_copy.instructionLines.text.replace(tmp_line, ""))
+                if parent_copy.instructionLines.text.endswith(","):
+                    parent_copy.instructionLines._setText(parent_copy.instructionLines.text[:-1])
+                parent_copy.instructionLines.set("count", str(
+                    int(parent_copy.instructionLines.get("count")) - 1))
+    except AttributeError:
+        pass
+    try:
+        for tmp_line in parent_copy.readPhaseLines.text.split(","):
+            if not line_contained_in_region(
+                    tmp_line,
+                    parent_copy.get("startsAtLine"),
+                    parent_copy.get("endsAtLine")):
+                parent_copy.readPhaseLines._setText(
+                    parent_copy.readPhaseLines.text.replace(tmp_line + ",", ""))
+                parent_copy.readPhaseLines._setText(
+                    parent_copy.readPhaseLines.text.replace(tmp_line, ""))
+                if parent_copy.readPhaseLines.text.endswith(","):
+                    parent_copy.readPhaseLines._setText(parent_copy.readPhaseLines.text[:-1])
+                parent_copy.readPhaseLines.set("count", str(
+                    int(parent_copy.readPhaseLines.get("count")) - 1))
+    except AttributeError:
+        pass
+    try:
+        for tmp_line in parent_copy.writePhaseLines.text.split(","):
```

Comment:
i see the last two try blocks are very similar to this one? can't u create a function and reuse?